pkg/internal/apis/certmanager/validation/issuer.go

```diff
@@ -253,16 +253,27 @@ func ValidateACMEChallengeSolverDNS01(p *cmacme.ACMEChallengeSolverDNS01, fldPat
 			el = append(el, field.Forbidden(fldPath.Child("azuredns"), "may not specify more than one provider type"))
 		} else {
 			numProviders++
-			el = append(el, ValidateSecretKeySelector(&p.AzureDNS.ClientSecret, fldPath.Child("azuredns", "clientSecretSecretRef"))...)
-			if len(p.AzureDNS.ClientID) == 0 {
-				el = append(el, field.Required(fldPath.Child("azuredns", "clientID"), ""))
+			// if ClientID is defined then clientSecret and tenantID must be defined
+			if len(p.AzureDNS.ClientID) > 0 {
+				el = append(el, ValidateSecretKeySelector(p.AzureDNS.ClientSecret, fldPath.Child("azuredns", "clientSecretSecretRef"))...)
+				if len(p.AzureDNS.TenantID) == 0 {
+					el = append(el, field.Required(fldPath.Child("azuredns", "tenantID"), ""))
+				}
 			}
+			// if ClientSecret is defined then both ClientID and TenantID must be defined
+			if p.AzureDNS.ClientSecret != nil {
```

Comment:
Hm, this logic seems a bit confusing. Can it be changed to be a single `if len(p.AzureDNS.ClientID) > 0 || p.AzureDNS.ClientSecret != nil {`? It'd then save us having the same validation code more than once :) unless there's something here I've missed that you are trying to account for

Reply:
done